Aztec/Classes/Libxml2/Converters/Out/OutHTMLConverter.swift

```diff
@@ -1,55 +1,215 @@
+        typealias Node              = Libxml2.Node
+        typealias TextNode          = Libxml2.TextNode
+        typealias CommentNode       = Libxml2.CommentNode
+        typealias RootNode          = Libxml2.RootNode
+
+        /// Indentation String to be applied
         ///
-        /// - Parameters:
-        ///     - rawNode: the Libxml2 Node to convert.
+        var indentationString = "  "
+
+        /// Indicates whether we want Pretty Print or not
         ///
-        /// - Returns: a String object representing the specified HTML data.
+        var prettyPrint = false
+
+
+        // MARK: - Initializers
+
+        init() {
+            // No Op
+        }
+
+        /// Converts a Node into it's HTML String Representation
         ///
         func convert(_ rawNode: Node) -> String {
+            return convert(node: rawNode).trimmingCharacters(in: CharacterSet.newlines)
+        }
+    }
+}
+
 
-            guard let outputBuffer = xmlAllocOutputBuffer(nil) else {
-                fatalError("This should not ever happen. Prevent the code from going further to avoid possible data loss.")
-            }
+// MARK: - Export: Nodes
+//
+private extension Libxml2.Out.HTMLConverter {
 
-            let xmlDocPtr = xmlNewDoc(nil)
+    /// Serializes a Node into it's HTML String Representation
+    ///
+    func convert(node: Node, level: Int = 0) -> String {
+        switch node {
+        case let node as RootNode:
+            return convert(root: node)
+        case let node as CommentNode:
+            return convert(comment: node)
+        case let node as ElementNode:
+            return convert(element: node, level: level)
+        case let node as TextNode:
+            return convert(text: node)
+        default:
+            fatalError("We're missing support for a node type.  This should not happen.")
+        }
+    }
 
-            defer {
-                xmlFreeDoc(xmlDocPtr)
-                xmlOutputBufferClose(outputBuffer)
-            }
+    /// Serializes a RootNode into it's HTML String Representation
+    ///
+    private func convert(root node: RootNode) -> String {
+        return node.children.reduce("") { (result, node) in
+            return result + convert(node: node)
+        }
+    }
 
-            let xmlNodePtr = Libxml2.Out.NodeConverter().convert(rawNode)
+    /// Serializes a CommentNode into it's HTML String Representation
+    ///
+    private func convert(comment node: CommentNode) -> String {
+        return "<!--" + node.comment + "-->"
+    }
 
-            xmlDocSetRootElement(xmlDocPtr, xmlNodePtr)
-            htmlNodeDumpFormatOutput(outputBuffer, xmlDocPtr, xmlNodePtr, "UTF-8", 0)
+    /// Serializes an ElementNode into it's HTML String Representation
+    ///
+    private func convert(element node: ElementNode, level: Int) -> String {
+        // Prefixes + Posfixes
+        let indentForOpeningTag = requiresOpeningTagPrefix(node) ? indentationString(for: level) : ""
```

Comment:
We're calling `requiresClosingTagPrefix()` and `requiresOpeningTagPrefix(node)` repeatedly on the same node.

Any chance we can store these calls in a single variable at the beginning of this method instead?